bin/setup.bat

```diff
@@ -15,6 +15,13 @@ goto finally
 
 :setup_jruby
 REM setup_java()
+if not defined JAVA_HOME IF EXIST %ProgramData%\Oracle\java\javapath\java.exe (
+    for /f "tokens=2 delims=[]" %%a in ('dir %ProgramData%\Oracle\java\javapath\java.exe') do @set JAVA_EXE=%%a
```

Comment:
Sorry if i wasn't clear. I think this line should be 

```
for /f "tokens=2 delims=[]" %%a in ('dir %ProgramData%\Oracle\java\javapath\java.exe') do @set JAVA_HOME=%%a
```

See https://docs.oracle.com/cd/E19182-01/820-7851/inst_cli_jdk_javahome_t/

> Set JAVA_HOME:
> 
> Right click My Computer and select Properties.
> 
> On the Advanced tab, select Environment Variables, and then edit JAVA_HOME to point to where the JDK software is located, for example, C:\Program Files\Java\jdk1.6.0_02.